source/class/qx/io/jsonrpc/protocol/Request.js

```diff
@@ -62,20 +69,41 @@ qx.Class.define("qx.io.jsonrpc.protocol.Request", {
       id = ++qx.io.jsonrpc.protocol.Request.__current_request_id;
     }
     this.set({ id });
-    this.__promise = new qx.Promise();
```

Comment:
you shouldnt need to do this - the only reason that `new qx.Promise()` exists is to support an externally accessible `resolve()` and `reject()`.

IE the previous code is correct, and your Request.resolve method becomes

```
resolve(value) {
      this.__promise.resolve(value);
    },
```

Although it seems strange that the `Request`'s promise is resolved/rejected  by `Client`?

My guess is that the reason there was a problem in the first place is because `Request.handleTransportException` rejected the promise and so does `Client.handleMessage`.  

That makes it sound like the two classes are arguing over who is responsible for the promise?  EG maybe only `Client` should resolve/reject?

Reply:
> you shouldnt need to do this - the only reason that new qx.Promise() exists is to support an externally accessible resolve() and reject().

You are right. This will make the fix even simpler. Local tests pass, let's see if the CI tests do.

> That makes it sound like the two classes are arguing over who is responsible for the promise? EG maybe only Client should resolve/reject?

Good point - I don't remember why I thought that a transport error should be handled at the level of the Request instead of the Client - maybe it should throw an exception and the Client should then handle it by rejecting the Request's promise. As this is a change of behavior, we would need to postpone it to qx 8.

Reply:
Problem is: The test I added doesn't seem to catch the problem raised in #10739. I just ran [testIssue10739()](https://github.com/cboulanger/qooxdoo/blob/Issue10739/source/class/qx/test/io/jsonrpc/Client.js#L342) with current master and the unhandled promise exception was not raised - so unfortunately, we are not smarter than before! :-( We need @bjorn-fischer to test whether the PR in its current form (i.e. by not using the qx.Promise API) helps in any way.